homework_03/src/main/java/org/application/App.java

```diff
@@ -1,9 +1,11 @@
 package org.application;
 public class App 
 {
-    public static void main( String[] args ) throws InterruptedException {
+    public static void main( String[] args ) {
         ValueCalculator calculator = new ValueCalculator();
```

Comment:
Perhaps I did something wrong, I don’t quite understand why 10 threads work slower than 1 and why for an array with 10 elements a large number of threads work faster compared to one thread than for an array with 1000000 elements

Reply:
You make this logic 
          int startIndex = i * segmentSize;
            int endIndex = (i == numOfThreads - 1) ? ARRAY_SIZE : (i + 1) * segmentSize;
            Float[] subArray = Arrays.copyOfRange(array, startIndex, endIndex);
not in thread internally
If you have 1 thread amount the code above will be called once
If you have 10 threads amount the code will be called 10 times

Reply:
Please, move the logic out of foreach and re-check perfomance